Added TypeScript definition file and test.

niktekusho opened the pull request:
Since tsd-check requires meow@5 which only works on Node >= 6, I decided to keep the definition tests to an optional npm task.
If @gakimball agrees to update the minimum Node version, then the optional task should be removed, and the TypeScript definition file check should be added to the default test npm task.

I kept the version of the module in [`package.json`](./package.json) equal to 1.0.0.
I think that a bump to 1.0.1 (patch) is enough to differentiate the new version from the currently released one, but I'll leave that up to you to decide.

Finally fixes #1 🎉

niktekusho committed:
:label: Added TypeScript definition file and test.

Since tsd-check requires meow@5 which only works on Node >= 6, I decided to keep the definition tests to an optional npm task. If the original author agrees to update the minimum Node version, then the optional task should be removed and the TypeScript definition file check should be added to the default test npm task.

gakimball commented:
Yeah, lets drop Node 4 and change the version to `2.0.0`. Only LTS for my packages when I can remember to update them 👌

niktekusho commented:
> 
> 
> Yeah, lets drop Node 4 and change the version to `2.0.0`. Only LTS for my packages when I can remember to update them 👌

I too use only Node's LTS versions. Next question: Node 4 -> 6 or 4 -> 8?
Node 6 is scheduled to EOL on 2019-04-01 so just a couple of months ahead...
Either the options I will update xo to the latest version.

niktekusho committed:
General updates.

- Drop Node@4 and Node@6
- Upgrade xo to latest version
- Lint fixes after xo upgrade
- TypeScript definition file test now runs on "npm test"

niktekusho committed:
Version 2.0.0

niktekusho commented:
I've just temporarily set the minimum Node version to 6.

gakimball commented:
This library is so small that the difference between Node 4/6/8 compatibility basically doesn't matter.

That being said, I think I'm fine doing minimum Node 8. Might as well not have to do this again in a few months haha.

niktekusho commented:
>I'm fine doing minimum Node 8. Might as well not have to do this again in a few months haha.

I was thinking exactly about that. 😆
